src/bench/block_assemble.cpp

```diff
@@ -42,8 +42,21 @@ static void AssembleBlock(benchmark::Bench& bench)
     }
 
     bench.run([&] {
-        PrepareBlock(test_setup->m_node, P2WSH_OP_TRUE);
+        PrepareBlock(test_setup->m_node, P2WSH_OP_TRUE, true);
+    });
+}
+static void BlockAssemblerAddPackageTxns(benchmark::Bench& bench)
+{
+    FastRandomContext det_rand{true};
+    auto testing_setup = MakeNoLogFileContext<TestChain100Setup>();
+    CTxMemPool& pool = *testing_setup.get()->m_node.mempool;
+    LOCK2(cs_main, pool.cs);
+    const auto transactions = testing_setup->PopulateMempool(det_rand, 1000, true);
+
+    bench.run([&] {
+        PrepareBlock(testing_setup->m_node, P2WSH_OP_TRUE, false);
     });
 }
 
 BENCHMARK(AssembleBlock, benchmark::PriorityLevel::HIGH);
+BENCHMARK(BlockAssemblerAddPackageTxns, benchmark::PriorityLevel::HIGH);
```

Comment:
this is what I get when trying to run the `BlockAssemblerAddPackageTxns` benchmark

Specs:
```
mac book pro 2019
2.3 GHz 8-Core Intel Core i9
16 GB 2667 MHz DDR4
AMD Radeon Pro 5500M 4 GB
```

Results:
```
./src/bench/bench_bitcoin -filter=BlockAssemblerAddPackageTxns --sanity-check
Running with --sanity-check option, benchmark results will be useless.

|               ns/op |                op/s |    err% |     total | benchmark
|--------------------:|--------------------:|--------:|----------:|:----------
|      813,714,108.00 |                1.23 |    0.0% |      0.81 | `BlockAssemblerAddPackageTxns`
```

Without `--sanity-check`:
```
./src/bench/bench_bitcoin -filter=BlockAssemblerAddPackageTxns

|               ns/op |                op/s |    err% |     total | benchmark
|--------------------:|--------------------:|--------:|----------:|:----------
|      830,079,779.00 |                1.20 |    1.4% |      9.16 | `BlockAssemblerAddPackageTxns`
```

Reply:
When compiled with `--enable-debug`, this slows down `make check` by ~30s for me. When run without `--sanity-check`, it takes ~280s. I think this needs to be a low priority benchmark.

Reply:
Yeah sorry, we had medium in the initial PR but it was removed later on, so agreed with changing it to low. We haven't defined any guidelines on when to downgrade to low afaik, but since devs run `make check` often, I think adding benches with meaningful (e.g. >1s?) `--sanity-check` overhead should run in low priority - unless their complexity/importance warrants otherwise.